curated_transformers/models/pytorch/attention.py

```diff
@@ -8,28 +8,37 @@
 from ...errors import Errors
 
 
-@dataclass
 class AttentionMask:
     bool_mask: Tensor
-    _logit_mask: Optional[Tensor] = None
+    _logit_mask: Optional[Tensor]
 
-    def __post_init__(self):
-        if self.bool_mask.dtype != torch.bool:
-            raise ValueError(Errors.E005.format(dtype=self.bool_mask.dtype))
+    def __init__(self, bool_mask: Tensor):
+        if bool_mask.dtype != torch.bool:
+            raise ValueError("Expected the attention mask to be of dtype 'torch.bool'")
```

Comment:
Let's add a comment specifying why we've inlined this error message (so that we won't inadvertently revert the change in the future).

Reply:
I wonder if we should instead document this in a developer doc. This will apply to any future place we add exceptions to Torch models. It seems better to have a checklist here than eventually having a comment on every exception raised in a module.

What do you think?

Reply:
Added `DEVELOP.md` now, let me know what you think.

Reply:
Good idea!